### PR TITLE
Tree_Preorder_Traversal.js added

### LINKED ISSUE #2550: Tree Preorder Traversal
- [x] C
- [x] C++
- [ ] C# - @sharmakeshav10
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [x] Go
- [x] Java
- [ ] JS - @afroz23
- [x] Kotlin - @PrachieNaik
- [x] Php
- [x] Python
- [ ] Readme - @SimranMakhija7
- [x] Ruby
- [ ] Rust
- [ ] TypeScript - @Shr03mink

### PR DESCRIPTION
**Pull request by afroz23**
Fixes #2550 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Tree_Preorder_Traversal.js added

#### Languages Used:
- javascript

#### Files Added:
- Tree_Preorder_Traversal.js 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/*
+output:
+Preorder Traversal:
+3
+1
+5
+4
+6
+*/
```
Comment (singh-shreya6) [inline Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js:93]: Please add blank line at the end of file.

File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+//Tree_Preorder_Traversal in javascript
+class Node
```
Comment (singh-shreya6) [inline Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js:2]: Please add a blank line before this.

File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+{
+  constructor(data)
```
Comment (singh-shreya6) [inline Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js:4]: Please add 4 space indentation at the start of each block.

Comment (afroz23) [reply Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js]: 4 space indentation Added

File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    }
+}
+class BinarySearchTree
+{
+    constructor()
+    {
+        this.root = null;
+    }
+//data inserting in tree
+insert(data)
```
Comment (singh-shreya6) [inline Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js:19]: All these subsequent function should be indented 4 spaces.

Comment (afroz23) [reply Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js]: Done mam :)

File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                this.insertNode(root.right, newNode); 
+            }
+        }
+    }
+
+    getRootNode()
+    {
+        return this.root;
+    }
+//function of preorder traversal
```
Comment (singh-shreya6) [inline Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js:64]: Please add 4 space indentation to this comment also.

Comment (afroz23) [reply Tree_Preorder_Traversal/Tree_Preorder_Traversal.js.js]: Done mam

**Comment by singh-shreya6** [comment]
Please squash your commits so that it can be merged.

**Comment by afroz23** [comment]
Mam, I am unable to squash all the commits, can I make a new pull request in which everything will be perfect enough to merge it :)

**Comment by singh-shreya6** [comment]
You can do that. But It is best that you learn to do it @afroz23, because in every PR you will be asked to squash commits. There are youtube videos also on how to squash commits.

**Pull request closed without merging**